app/models/throttle.rb

```diff
@@ -54,23 +54,35 @@ def throttled?
 
   def remaining_count
     return 0 if throttled?
-    max_attempts, _attempt_window_in_minutes = Throttle.config_values(throttle_type)
-    max_attempts - attempts
+    Throttle.max_attempts(throttle_type) - attempts
+  end
+
+  def expires_at
+    return Time.zone.now if attempted_at.blank?
+    attempted_at + Throttle.attempt_window_in_minutes(throttle_type).minutes
   end
 
   def expired?
-    return true if attempted_at.blank?
-    _max_attempts, attempt_window_in_minutes = Throttle.config_values(throttle_type)
-    attempted_at + attempt_window_in_minutes.to_i.minutes < Time.zone.now
+    expires_at <= Time.zone.now
   end
 
   def maxed?
-    max_attempts, _attempt_window_in_minutes = Throttle.config_values(throttle_type)
-    attempts >= max_attempts
+    attempts >= Throttle.max_attempts(throttle_type)
+  end
+
+  def self.attempt_window_in_minutes(throttle_type)
+    THROTTLE_CONFIG.with_indifferent_access.dig(throttle_type, :attempt_window)
```

Comment:
WDYT of just making the `THROTTLE_CONFIG` hash have indifferent access?

```
THROTTLE_CONFIG = {
  # ... etc
}.with_indifferent_access.freeze
```

Reply:
> WDYT of just making the `THROTTLE_CONFIG` hash have indifferent access?

I like it! Originally I tried removing the `with_indifferent_access` altogether, but quickly discovered it broke a lot of existing code.

Reply:
> WDYT of just making the `THROTTLE_CONFIG` hash have indifferent access?

Updated in the rebased bd57718.